sys/random/Kconfig

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+menuconfig MODULE_RANDOM
+    bool "Pseudo-Random Number Generation"
+    depends on TEST_KCONFIG
+    select MODULE_LUID
+    select MODULE_PERIPH_HWRNG if HAS_PERIPH_HWRNG && !MODULE_PUF_SRAM
```

Comment:
> I'm under the impression this dependency is incorrectly modeled in master though

Right, I just check the code and LUID is used only for the auto init.

> Also wouldn't this be better modeled with a choice?

It would be good if the user can select the seed source explicitly. So something like

```
choice
    bool "Seed source"
    depends on MODULE_AUTO_INIT

config MODULE_RANDOM_SEED_SOURCE_PUF
    bool "PUF-SRAM"
    depends on MODULE_PUF_SRAM

config MODULE_RANDOM_SEED_SOURCE_PERIPH_HWRNG
    bool "Hardware RNG"
    depends on HAS_PERIPH_HWRNG
    select MODULE_PERIPH_HWRNG

config MODULE_RANDOM_SEED_SOURCE_PERIPH_CPUID
    bool "CPU ID"
    select MODULE_LUID

config MODULE_RANDOM_SEED_SOURCE_DEFAULT
    bool "Default Seed"

endchoice
```

Should I make the changes in random (only makefile + code) on a separate PR?

Reply:
It might make sense to adapt those in a follow up, currently the PR is aligned with master so we can get it in and the it becomes a smaller change (that some may object?)

Reply:
> It would be good if the user can select the seed source explicitly. So something like

+1

Reply:
As discussed offline this PR is modeling Kconfig as it is in master, IMO we can go with and keep discussing improvements on another PR